packages/@aws-cdk/aws-apigatewayv2-integrations/lib/http/service-discovery.ts

```diff
@@ -32,12 +22,16 @@ export class HttpServiceDiscoveryIntegration extends HttpPrivateIntegration {
   }
 
   public bind(_: HttpRouteIntegrationBindOptions): HttpRouteIntegrationConfig {
+    if (!this.props.vpcLink) {
+      throw new Error('The vpcLink property is mandatory');
+    }
+
     return {
       method: this.props.method ?? this.httpMethod,
       payloadFormatVersion: this.payloadFormatVersion,
       type: this.integrationType,
       connectionType: this.connectionType,
-      connectionId: this.props.vpcLink.vpcLinkId,
+      connectionId: this.props.vpcLink?.vpcLinkId,
```

Comment:
The optional check with `?` should not be required since we are throwing before for it.